### PR TITLE
Added SELinux and NFS exports section

### PR DESCRIPTION
**Pull request by markturansky**
@sosiouxme Please review these security settings for SELinux and NFS?

### CODE REVIEW

File: admin_guide/persistent_storage_nfs.adoc [hunk 1]
```diff
@@ -84,3 +84,18 @@ reclamation of discarded resources can be configured for each persistent volume.
...
+
+By default, SELinux will disallow a connection from a pod to a remote NFS server.  Enable this in SELinux on each node with the following command:
+
+```
+setsebool virt_use_nfs 1
+```
+
+Additionally, each exported volume on the NFS server itself must conform to the following:
+
+* each export must be: ```(rw,sync,all_squash)```
```
Comment (rootfs) [inline admin_guide/persistent_storage_nfs.adoc]: `sync` is probably optional. Also would be clear if we also disable network level access control, e.g. 
`/exported_fs *(rw,no_root_squash)`

Comment (markturansky) [inline admin_guide/persistent_storage_nfs.adoc]: thanks @rootfs.  that matches what @jsafrane has in his NFS server pod image.

doc updated.

File: admin_guide/persistent_storage_nfs.adoc [hunk 2]
```diff
@@ -84,3 +84,18 @@ reclamation of discarded resources can be configured for each persistent volume.
 Many ways to script these tasks exist.  We've created an link:https://github.com/openshift/openshift-ansible/tree/master/roles/kube_nfs_volumes[Ansible playbook] to help you get started.
 
 
+== SELinux and NFS export settings
+
+By default, SELinux will disallow a connection from a pod to a remote NFS server.  Enable this in SELinux on each node with the following command:
```
Comment (sosiouxme) [inline admin_guide/persistent_storage_nfs.adoc]: It's not really true that the connection is disallowed. What I saw happen was that the pod wasn't allowed to write to an NFS mount. So without this, you would actually mount up the volume fine. The app would just fail when it tried to actually write some data.

File: admin_guide/persistent_storage_nfs.adoc [hunk 3]
```diff
@@ -84,3 +84,18 @@ reclamation of discarded resources can be configured for each persistent volume.
...
+
+```
+setsebool virt_use_nfs 1
+```
+
+Additionally, each exported volume on the NFS server itself must conform to the following:
+
+* each export must be: ```/example_fs *(rw,no_root_squash)```
```
Comment (sosiouxme) [inline admin_guide/persistent_storage_nfs.adoc]: While * is simplest to tell people, without knowledge of their networking, we should note that it is fine to restrict it to the range of node hosts that could possibly be using the storage. Might be best just to give an example and refer to the `/etc/exports` man page.

I found that no_root_squash was not sufficient in the mysql case - it got confused by the ownership of the volume itself. I'm not certain of the best recommendation here but what seemed to work was `all_squash`.

So I'd say something like:

> To enable use by applications with arbitrary user permissions, each export should have the permissions `(rw,all_squash)`, e.g. in `/etc/exports`:
> 
>         /example_fs  *(rw,all_squash)
> 
> Refer to the `/etc/exports` man page for further details of defining exports, e.g. to restrict access to only node hosts.

File: admin_guide/persistent_storage_nfs.adoc [hunk 4]
```diff
@@ -84,3 +84,18 @@ reclamation of discarded resources can be configured for each persistent volume.
...
+* each export is owned by nfsnobody: ```chmod nfsnobody:nfsnobody```
+* each export must have the permissions: ```chmod 700```
+
+Disabling SELinux entirely with ```setenforce 0``` would also work but is generally discouraged.
```
Comment (sosiouxme) [inline admin_guide/persistent_storage_nfs.adoc]: Wouldn't even mention this. It's as easy to copy/paste `setsebool virt_use_nfs 1` as this.

File: admin_guide/persistent_storage_nfs.adoc [hunk 5]
```diff
@@ -84,3 +84,18 @@ reclamation of discarded resources can be configured for each persistent volume.
 Many ways to script these tasks exist.  We've created an link:https://github.com/openshift/openshift-ansible/tree/master/roles/kube_nfs_volumes[Ansible playbook] to help you get started.
 
 
+== SELinux and NFS export settings
+
+By default, SELinux will disallow a connection from a pod to a remote NFS server.  Enable this in SELinux on each node with the following command:
+
+```
+setsebool virt_use_nfs 1
```
Comment (sosiouxme) [inline admin_guide/persistent_storage_nfs.adoc]: `setsebool -P` (to make it persist after reboot)

**Comment by markturansky** [comment]
@sosiouxme thanks!  that was helpful.  Feedback addressed.

**Comment by adellape** [comment]
@markturansky If you're set, I'll merge and do a pass on minor formatting.

**Comment by markturansky** [comment]
@adellape I'll let @sosiouxme have the final sign-off, as he understands these issues better than I do.

**Comment by sosiouxme** [comment]
LGTM

**Comment by danjeffries1** [comment]
The current iteration of this does not seem to work

setsebool -P virt_use_nfs 1 1   

That throws and error.  The following seems to work:

setsebool -P virt_use_nfs=1

**Comment by sosiouxme** [comment]
@danjeffries1 True... could you make an issue or a PR?

**Comment by sosiouxme** [comment]
Actually I just created one.

https://github.com/openshift/openshift-docs/issues/772

**Pull request merged**